controllers/managedocs_controller_test.go

```diff
@@ -489,6 +489,7 @@ var _ = Describe("ManagedOCS controller", func() {
 			} else {
 				Expect(k8sClient.Create(ctx, pdSecret)).Should(Succeed())
 			}
+			utils.WaitForResource(k8sClient, ctx, pdSecret, timeout, interval)
```

Comment:
This doesn't make sense, what exactly are we waiting for? In both Update and Create k8s guarantee to return a resource if Create or Update is completed successfully. 
If this solves the problem it is just because this will create a sort of delay (sleep) which mean we do not know what the underlaying issue is.

Let's figure out what is the actual issue and solve it.